pandas/indexes/multi.py

```diff
@@ -71,8 +71,10 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
                 name=None, **kwargs):
 
         # compat with Index
-        if name is not None:
-            names = name
+        if name is None and hasattr(levels, 'name'):
+            name = levels.name
+        if isinstance(levels, MultiIndex):
+            return levels.copy(name=name, names=names, deep=copy)
```

Comment:
I assumed we want ``idx.__class__(idx)`` to return a copy of ``idx`` for any index ``idx``. But maybe it was just my fantasy...